File: algorithms/11-ContainerWithMostWater/Solution.java
class Solution {
    public int maxArea(int[] height) {
        if(height==null || height.length==0){
            return 0;
        }
        int i=0;
        int j=height.length-1;
        int result =0;
        while (i <j){
            int x = height[i];
            int y = height[j];
            int temp = Math.min(x,y)*(j-i);
            if(temp > result){
                result= temp;
            }
            if (height[i] < height[j]){
                i++;
            }    
            else if (height[i] > height[j]){
                j--;
            }else{
                i++;j--;
            }
                
        }
        return result;
    }
}

File: algorithms/11-ContainerWithMostWater/Solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        if not height or len(height)==0:
            return 0
        i=0
        j=len(height)-1
        result =0
        while i <j:
            x = height[i]
            y = height[j]
            result = max(min(x,y)*(j-i) ,result)
            if height[i] < height[j]:
                i=i+1
            elif height[i] > height[j]:
                j=j-1
            else:
                i=i+1
                j=j-1

        return result
                         
            
